From e06980efe8c3364a774674a3d29c361f773ecd0c Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Fri, 23 Jan 2026 19:45:45 +0100
Subject: [PATCH] add new system and io_uring permission definitions

Add new permission definitions in the system and io_uring classes that
have been introduced in newer kernels, but not yet incorporated into the
Fedora SELinux policy. Similar to the SELinux project's reference
policy, only define the permissions, but do not yet introduce any policy
for them.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/flask/access_vectors | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/policy/flask/access_vectors b/policy/flask/access_vectors
index 8ec095851..70ac2f35a 100644
--- a/policy/flask/access_vectors
+++ b/policy/flask/access_vectors
@@ -444,6 +444,12 @@ class system
 	syslog_console
 	module_request
 	module_load
+	firmware_load
+	kexec_image_load
+	kexec_initramfs_load
+	policy_load
+	x509_certificate_load
+
 	# these are overloaded userspace
 	# permissions from systemd
 	halt
@@ -1106,6 +1112,7 @@ class io_uring
     override_creds
     sqpoll
     cmd
+    allowed
 }
 
 class user_namespace
-- 
2.52.0

